lib/substitutions.js

```diff
@@ -0,0 +1,50 @@
+"use strict";
+
+// Ordered
+var substitutionPairs = [];
+
+function add(slack, matrix) {
+    substitutionPairs.push({
+        slack: slack,
+        matrix: matrix
+    });
+}
+
+// slack -> matrix substitutions are performed top -> bottom
+// matrix -> slack substitutions are performed bottom -> top
+add("&lt;", "<");
+add("&gt;", ">");
+add("&amp;", "&"); // &amp; must be after all replacements involving &s.
+
+/**
+ * Performs any escaping, unescaping, or substituting required to make the text
+ * of a Slack message appear like the text of a Matrix message.
+ *
+ * @param {string} string the text, in Slack's format.
+ */
+var slackToMatrix = function(string) {
+    for (var i = 0; i < substitutionPairs.length; ++i) {
+        var pair = substitutionPairs[i];
+        string = string.replace(pair.slack, pair.matrix);
```

Comment:
Ah yes, `String.replace` is special. This will only actually replace the *first occurrence* of `pair.slack` - You can get around this by specifying a global regex e.g. `string.replace(/foo/g, "bar")` or if you're regex averse.. well.. keep applying it until there are no diffs? :D